### PR TITLE
[js/webgpu] Enable the NCHW ConvMatMul path

### PR DESCRIPTION
**Pull request by qjia7**
1) Enable pointwise NCHW conv2d by MatMul.
2) Enable non-pointwise NCHW conv2d by convMatMul.
3) Fix bug when `sameSize` is true

### CODE REVIEW

**Comment by qjia7** [comment]
@guschmue @fs-eire Please take a look, thanks.

**Review by qjia7 — COMMENTED** [review]
The mobilenetv2-12 with NCHW data format becomes 12 ms from 51 ms with this change on my device, which is on par to NHWC data format.
I think it's a good perf reference for NHWC when we do optimizations.
Note that the convMatmul with vec4 path for non-pointwise conv2d is still not enabled. So if the models heavily use non-pointwise conv2d, the NCHW perf is still worse than NHWC. I will try to optimize the vec4 path in separate PRs.

File: js/web/lib/wasm/jsep/webgpu/ops/3rd-party/conv2d_mm_webgpu.ts [hunk 1]
```diff
@@ -160,17 +160,14 @@ export const createConv2DMatMulProgramInfo =
...
 
       // TODO: fine tune size
       const dispatchX = isChannelsLast ? outChannels : outWidth * outHeight;
       const dispatchY = isChannelsLast ? outWidth * outHeight : outChannels;
-      const workGroupSize: [number, number, number] =
-          isVec4 ? [8, 8, 1] : [dispatchX <= 4 ? 4 : 16, dispatchX > 4 && dispatchY <= 4 ? 4 : 16, 1];
-      const elementsPerThread =
-          isVec4 ? [4, 4, 1] : [dispatchX <= 4 ? 1 : 2, dispatchX > 4 && dispatchY <= 4 ? 1 : 2, 1];
+      const workGroupSize: [number, number, number] = [8, 8, 1];
+      const elementsPerThread = dimAOuter <= 8 ? [4, 1, 1] : [4, 4, 1];
```
[the remark this answers is not in the document]
Comment (dakenf) [reply js/web/lib/wasm/jsep/webgpu/ops/3rd-party/conv2d_mm_webgpu.ts:170]: Can confirm this works and works much faster. Before these changes to workgroup size i had error `tileAWidth 4 must be divisible by workgroupSize[0] 16`

**Review by satyajandhyala — APPROVED** [review]
ConvTranspose also uses matmul_packed_webgpu.ts, in conv_backprop_mm_webgpu.ts. I think it is good if we can add a ConvTranspose test cases that exercises newly added code.

**Review by satyajandhyala — APPROVED** [review]

**Review by guschmue — APPROVED** [review]

**Pull request merged**